### PR TITLE
docs: Improve previous version navigation

### PR DESCRIPTION
**Pull request by JeanMeche**
In archive mode the link to the current doc will point to same page. If the page doesn't exist anymore, it will show a contextualized warning message.

See #46850

## PR Type
What kind of change does this PR introduce?

- [x] Documentation content changes

This PR was previously at #48965

### CODE REVIEW

**Review by hexiaokang — APPROVED** [review]

**Review by josephperrott — APPROVED** [review]
LGTM, but @mgechev will need to decide if we want to do it.

**Comment by jessicajaniuk** [comment]
@JeanMeche Looks like this needs a rebase

File: aio/src/app/layout/mode-banner/mode-banner.component.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
+
```
Comment (jessicajaniuk) [inline aio/src/app/layout/mode-banner/mode-banner.component.ts:1]: Were these extra empty lines intentional?

**Review by jessicajaniuk — APPROVED** [review]
Other than the need for a rebase, this looks good. Once you rebase, I'll approve for the size-tracking changes.

**Comment by jessicajaniuk** [comment]
@JeanMeche Looks like AIO payloads have changed since the pre-rebase run.

**Comment by JeanMeche** [comment]
@jessicajaniuk  It's OK now, `test_win` is still flaky

**Comment by jessicajaniuk** [comment]
@JeanMeche The `test_win` issue seems to be affecting everyone. Seems to be an infra issue, and not this PR.

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: docs-infra, size-tracking

**Comment by alan-agius4** [comment]
@JeanMeche, can you please fix the conflict?

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit ed8a595408615919c55dbe1200748592119f5641.

**Pull request closed without merging**